gp-includes/things/translation.php

```diff
@@ -417,24 +417,79 @@ public function for_translation( $project, $translation_set, $page, $filters = a
 
 		$where = implode( ' AND ', $where );
 		if ( $where ) {
-			$where = 'AND '.$where;
+			$where = 'AND ' . $where;
 		}
 
+		$where = 'o.project_id = ' . (int) $project->id . ' AND o.status = "+active" ' . $where;
+
 		$join_where = implode( ' AND ', $join_where );
 		if ( $join_where ) {
-			$join_where = 'AND '.$join_where;
+			$join_where = 'AND ' . $join_where;
 		}
 
-		$sql_sort = sprintf( $sort_by, $sort_how );
+		$fields = array(
+			't.*',
+			'o.*',
+			't.id as id',
+			'o.id as original_id',
+			't.status as translation_status',
+			'o.status as original_status',
+			't.date_added as translation_added',
+			'o.date_added as original_added',
+		);
+
+		$join = "$join_type JOIN {$wpdb->gp_translations} AS t ON o.id = t.original_id AND t.translation_set_id = " . (int) $translation_set->id;
+
+		$orderby = sprintf( $sort_by, $sort_how );
 
 		$limit = $this->sql_limit_for_paging( $page, $this->per_page );
 
+		/**
+		 * Filters the 'for_translation' query SQL clauses.
+		 *
+		 * @since 2.3.0
+		 *
+		 * @param array              $pieces          {
+		 *     Translation query SQL clauses.
+		 *
+		 *     @type array  $fields     Fields to select in the query.
+		 *     @type string $join       JOIN clause of the query.
+		 *     @type string $join_where Conditions for the JOIN clause.
```

Comment:
@yoavf @akirk What do you think about renaming `$join_where` to `$join_on`? It's not really a WHERE clause because it's added before `WHERE`:

```sql
SELECT SQL_CALC_FOUND_ROWS
t.*, o.*, t.id as id, o.id as original_id, t.status as translation_status, o.status as original_status, t.date_added as translation_added, o.date_added as original_added
FROM wp_gp_originals as o
LEFT JOIN wp_gp_translations AS t ON o.id = t.original_id AND t.translation_set_id = 6 AND t.status != "rejected" AND t.status != "old" AND (t.status = 'current' OR t.status = 'waiting' OR t.status = 'fuzzy')
WHERE o.project_id = 3 AND o.status = "+active"
ORDER BY o.date_added DESC
LIMIT 100 OFFSET 0
```